blaze/expr/collections.py

```diff
@@ -418,6 +418,7 @@ def on_right(self):
         return on_right
 
     @property
+    @memoize
     def schema(self):
```

Comment:
@llllllllll adding `__weakref__` to `Join.__slots__` led to this cryptic error:

```
blaze/__init__.py:19: in <module>
    from .expr import (Symbol, TableSymbol, symbol, ndim, shape)
blaze/expr/__init__.py:11: in <module>
    from .collections import *
blaze/expr/collections.py:359: in <module>
    class Join(Expr):
E   TypeError: Error when calling the metaclass bases
E       __weakref__ slot disallowed: either we already got one, or __itemsize__ != 0
```

I'm also curious about the `__slots__` usage in general in `collections.py`, see #1268.